src/rdfTypes/GeometryInfoHelpersImpl.h

```diff
@@ -190,6 +192,76 @@ inline util::geo::DBox projectInt32WebMercToDoubleLatLng(
           projectInt32WebMercToDoubleLatLng(box.getUpperRight())};
 };
 
+// Compute the length of the outer boundary of a polygon.
+inline double computeMetricLengthPolygon(const Polygon<CoordType>& geom) {
+  return latLngLen<CoordType>(geom.getOuter());
+}
+
+// Compute the length of a multi-geometry by adding up the lengths of its
+// members.
+template <typename T>
+inline double computeMetricLengthMulti(
+    const std::vector<T>& geom, std::function<double(const T&)> lenFunction) {
+  return ::ranges::accumulate(::ranges::transform_view(geom, lenFunction), 0);
+}
```

Comment:
Either use `absl::FunctionRef` (to avoid allocations etc), or directly make the function also a template parameter.
(std::function has possible overheads on many levels).